spring-cloud-vault-core/src/main/java/org/springframework/cloud/vault/VaultClient.java

```diff
@@ -170,6 +291,10 @@ private HttpHeaders createHeaders(VaultToken vaultToken) {
 		return headers;
 	}
 
+	private HttpHeaders createHeaders() {
```

Comment:
`VaultClient` is login-method agnostic. All state and login features should be kept inside `VaultTemplate`. We support various login methods that keep their state in a `VaultState` class. It's not intended to write the token back into `VaultProperties`.